packages/core-web/src/index.js

```diff
@@ -21,50 +21,61 @@ function scrollToUrlAnchorHeading() {
   }
 }
 
-function detectAndApplyFixedHeaderStyles() {
+function detectAndApplyHeaderStyles() {
   jQuery(':header').each((index, heading) => {
     if (heading.id) {
       jQuery(heading).removeAttr('id'); // to avoid duplicated id problem
     }
   });
 
-  const headerSelector = jQuery('header[fixed]');
-  const isFixed = headerSelector.length !== 0;
-  if (!isFixed) {
+  const headerSelector = jQuery('header[sticky]');
+  if (headerSelector.length === 0) {
     return;
   }
 
-  window.addEventListener('resize', () => {
-    if (window.innerWidth > 767 && headerSelector.hasClass('hide-header')) {
-      headerSelector.removeClass('hide-header');
-      headerSelector.css('overflow', '');
-    }
-  });
-
-  const updateHeaderHeight = () => {
+  function updateHeaderHeight() {
     const newHeaderHeight = headerSelector.height();
-    document.documentElement.style.setProperty('--header-height', `${newHeaderHeight}px`);
-  };
+    document.documentElement.style.setProperty('--sticky-header-height', `${newHeaderHeight}px`);
+  }
 
-  const toggleHeaderOverflow = () => {
-    const headerMaxHeight = headerSelector.css('max-height');
+  let isHidden = false;
+  function showHeader() {
+    isHidden = false;
+    headerSelector.removeClass('hide-header');
+  }
+  headerSelector[0].addEventListener('transitionend', () => {
     // reset overflow when header shows again to allow content
     // in the header such as search dropdown etc. to overflow
-    if (headerMaxHeight === '100%') {
```

Comment:
most of the changes here (resizeObserver, this line, isHidden) are necessary due to switching to `transform translate` for hiding the header (height change dosen't work with `position: sticky`), which should be simpler (I think) to deal with too.